script.domoticz.scenes/resources/lib/script.py

```diff
@@ -0,0 +1,307 @@
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    if useSsl == True:
+        protocol = 'https://'
+    else:
+        protocol = 'http://'
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=devices&used=true&filter=all&favorite=1'
+    try:
+        result = requests.get(url)
+    except:
+        return -1
+
+    answer = result.content
+
+    jsonResult = json.loads(answer)
+    deviceResult = jsonResult['result']
+    return(deviceResult)
+
+
+def get_all_devices_dict(host, port, useSsl, username=None, password=None):
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    if useSsl == True:
+        protocol = 'https://'
+    else:
+        protocol = 'http://'
+
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=devices&used=true&displayhidden=1'
+    try:
+        result = requests.get(url)
+    except:
+        return -1
+
+    answer = result.content
+
+    jsonResult = json.loads(answer)
+    deviceResult = jsonResult['result']
+
+    return(deviceResult)
+
+
+def get_version(host, port, useSsl, verifySsl, username=None, password=None):
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    if useSsl == True:
+        protocol = 'https://'
+    else:
+        protocol = 'http://'
+
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=command&param=getversion'
+    try:
+        result = requests.get(url, verify=verifySsl)
+    except:
+        return -1
+
+    answer = result.content
+    jsonResult = json.loads(answer)
+    return(jsonResult)
+
+
+def get_all_switches(host, port, useSsl, username=None, password=None):
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    if useSsl == True:
+        protocol = 'https://'
+    else:
+        protocol = 'http://'
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=devices&filter=light&used=true&order=Name'
+
+    try:
+        result = requests.get(url)
+    except:
+        return -1
+
+    answer = result.content
+
+    jsonResult = json.loads(answer)
+    deviceResult = jsonResult['result']
+    return(deviceResult)
+
+
+def switch_scene(host, port, idx, useSsl=False, username=None, password=None):
+    if useSsl is True:
+        base_url = "https://" + host + ":" + str(port)
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchscene&idx=" + str(idx) + "&switchcmd=On"
```

Comment:
You have loads of code repetition. Wouldn't it make more sense to create a common method and only vary the argument passed to the URL?

Reply:
Cleaned up